framework/doc/content/syntax/LinearFVBCs/index.md

```diff
@@ -0,0 +1,103 @@
+applied in different manners for different terms in the partial differential equation.
+This means that the `LinearFVBCs` only provide objects to specify these boundary quantities,
+and would not contribute to the system matrix and right hand side directly (only through kernels).
+
+For more information on general design choices in this setting [click here](/linear_fv_design.md)
+
+## LinearFVBCs block
+
+FVM boundary conditions are added to simulation input files in the `LinearFVBCs` as in the example below.
+
+!listing test/tests/linearfvkernels/advection/advection-1d.i
+         block=LinearFVBCs
+         caption=Example of the LinearFVBCs block in a MOOSE input file.
+
+In this example input, an advection equation with Dirichlet boundary condition on the left and outflow boundary conditions on the right is solved. To understand the differences between
+these two boundary conditions, let's start with the advection equation:
+
+\begin{equation}
+  \nabla \cdot (\vec{v} u) = S,
+\end{equation}
+
+with $\vec{v}$ denoting the velocity vector, $u$ the solution, and $S$ a potentially space-dependent
+source term. The boundary condition on the left can be expressed as:
+
+\begin{equation}
+  u_b = f(x_b),
+\end{equation}
+
+while the outflow boundary expresses outward advection with the solution value
+on the boundary and a predefined velocity.
+
+Both boundary conditions can be applied in an integral sense through the discretized
+advection term on the cell adjacent to the boundary:
+
+\begin{equation}
+  \int\limits_{V_b} \nabla \cdot (\vec{v} u) dV \approx \left(\sum\limits_i \vec{n}_i
+  \cdot \vec{v}_i u_{f,i}|S_i|\right) + \vec{n}_b \cdot \vec{v}_b u_b |S_b|~,
+\end{equation}
+
+where the $i$ index denotes internal faces of the cell, while $b$ denotes the only face on the boundary.
+This means that the only thing we need to supply to this formula is a way to compute the contributions to
+the system matrix and right hand side from the boundary value $u_b$.
+For example for the Dirichlet boundary $u_b = f(x)$,
+while for the outflow boundary it can be either the the cell centroid value ($u_b = u_C$)
+or an extrapolated value. This also means that the Dirichlet boundary contributes to
+the right hand side of the system only, whereas the outflow boundary condition can contribute to both.
+
+## Functions to override:
+
+Different linear finite volume kernels might use the quantities provided by these boundary
+conditions differently, but there are some common functionalities which are used more
+frequently between diffusion and advection kernels.
+The following functions represent this common functionality:
+
+- `computeBoundaryValue` computes the boundary value of the field.
+- `computBoundaryNormalGradient` computes the normal gradient of the variable on this boundary.
+- `computeBoundaryValueMatrixContribution` computes the matrix contribution that would come from
```

Comment:
No Jacobian in these linear problems

Reply:
do we have another name for this matrix then?
I m not clear on why we cant view this matrix as a Jacobian. Isnt it still the derivative of the residual of the equation?  (with a lot of terms chosen to be explicit)

Reply:
A user of these linear systems will likely not describe their problem in a residual fashion such that the linearization is

$\hat{J} \vec{\delta u} = -\vec{R}$

Instead they will write it directly as

$\hat{A} \vec{u} = \vec{b}$

Someone coming from a linear system paradigm may not even know what a residual is, e.g. that they need to move all terms to one side (canonically the left-hand side) and consider source terms which are independent of the degrees of freedom as part of the residual (left-hand side). They probably only think of such terms as belonging on the *right-hand side*.

Reply:
I've personally never come across CFD people who spend much time thinking in the linear system paradigm (I've seen Stokes talked about for just long enough to explain "saddle-point problems" and their stability before professors whip out the nonlinear convection term and *its* stability), but I figured they must exist, so I started searching around to see what they call their matrix.

"Stiffness matrix".  No elasticity in the problem in these papers, not even *finite elements* in the problem in many of these papers, and they still call the Jacobian the "stiffness" matrix.  What the hell.

Ranting aside, even if A is always going to be either the Jacobian or its negative (depending on which hypothetical convention we might have used for constructing a residual), that sign convention is tricky.  I think I've seen "b-Au" used more often than "Au-b" when squeezing linear problems into nonlinear solvers, and for people thinking in those terms A is the negative of the Jacobian, so calling it "Jacobian" instead of just "Matrix" might be confusing.

Reply:
Sign aside it s still the same matrix in my mind. 
Au = b
Au - Au_prev = b - A u_prev
A (delta u) = - residual_prev

ok for not using Jacobian if it's going to be unclear.
Looks like stiffness matrix isnt a top candidate either.
Thoughts about calling it the A matrix? so in the routine names: matrix -> AMatrix
then RHS can also be called b-vector (or not if RHS is actually clearer)

Reply:
RHS is very standard terminology in my experience, though I may be biased. - Roy H. Stogner

Come to think about it, "LHS matrix" might be an improvement over just "matrix".

Reply:
LHSMatrix seems good to me

Reply:
> Personally I'm content with the terminology used here

To be clear, so am I; though I do think qualifying with "LHS" is a slight improvement.

Reply:
I like either matrix or SystemMatrix

Reply:
Okay, then I'm leaving it as is.